pin/pin.go

```diff
@@ -246,6 +241,10 @@ func (p *pinner) Pin(ctx context.Context, node ipld.Node, recurse bool) error {
 
 		p.recursePin.Add(c)
 	} else {
+		if p.recursePin.Has(c) {
+			return fmt.Errorf("%s already pinned recursively", c.String())
+		}
+
```

Comment:
This is also done in the current pinner:
https://github.com/ipfs/go-ipfs-pinner/blob/master/dspinner/pin.go#L246-L247